eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -460,18 +460,43 @@ jobs:
 
   # Only upload test results if enabled
   - ${{ if eq(parameters.runTests, 'True') }}:
+    - ${{ if eq(parameters.buildSourceOnly, 'True') }}:
+      - task: PublishTestResults@2
+        displayName: Publish Source Build Smoke Test Results
+        condition: succeededOrFailed()
+        continueOnError: true
+        inputs:
+          testRunner: VSTest
+          testResultsFiles: 'test/**/*.trx'
+          searchFolder: $(sourcesPath)
+          mergeTestResults: true
+          publishRunAttachments: true
+          testRunTitle: SourceBuild_SmokeTests_$(Agent.JobName)
+
     - task: PublishTestResults@2
       displayName: Publish Test Results
       condition: succeededOrFailed()
       continueOnError: true
       inputs:
-        testRunner: vSTest
+        testRunner: VSTest
         testResultsFiles: 'artifacts/TestResults/Release/*.trx'
         searchFolder: $(sourcesPath)
         mergeTestResults: true
         publishRunAttachments: true
         testRunTitle: Tests_$(Agent.JobName)
 
+    - task: PublishTestResults@2
+      displayName: Publish Scenario Test Results
+      condition: succeededOrFailed()
+      continueOnError: true
+      inputs:
+        testRunner: xUnit
+        testResultsFiles: 'artifacts/TestResults/**/*.xml'
```

Comment:
There might be other xml files in that directory already today or eventually in the future that aren't tied to scenario tests. It think it would make sense to introduce a subfolder for scenario-tests, i.e. `artifacts/TestResults/Release/scenario-tests`.

Reply:
Agreed that the tests results should be separated.  This includes the artifacts gathered in `Prepare BuildLogs staging directory and check assets`

Reply:
Fixed in https://github.com/dotnet/installer/pull/19222/commits/7bb0e4a3be844601edee86c6f66d76a702a8a056